digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala

```diff
@@ -375,20 +392,92 @@ class SpeedLimitsTierekisteriImporter extends LinearAssetTierekisteriImporterOpe
+      }
+    }
+
+    trUrbanAreaAssets.filter(_.startRoadPartNumber == addressSection.roadPartNumber).foreach(
+      trUA =>
+        if ((trUA.startAddressMValue <= addressSection.startAddressMValue) && (trUA.endAddressMValue >= addressSection.endAddressMValue.head)) {
+          return createSpeedLimitByUrbanArea(trUA.assetValue, addressSection)
+        } else {
+          createMultiSpeedLimitInAddressSection(trUA, roadLink, addressSection, roadAddress, roadSide)
+          return None
+        }
+    )
+    Some(TierekisteriTrafficSignData(addressSection.roadNumber, addressSection.roadPartNumber, addressSection.roadPartNumber, addressSection.track, addressSection.startAddressMValue, addressSection.startAddressMValue, roadSide, TRTrafficSignType.EndUrbanArea, defaultSpeedLimit.toString))
+  }
+
+  protected def createMultiSpeedLimitInAddressSection(trUAAsset: TierekisteriUrbanAreaData, roadLink: VVHRoadlink,
+                                                      addressSection: AddressSection, roadAddress: ViiteRoadAddress,
+                                                      roadSide: RoadSide) = {
+    def verifyDefaultTolerance(urbanAreaAddressMValue: Double, addressSectionAddressMValue: Double): Boolean ={
+      val defaultTolerance = 2
+      ((urbanAreaAddressMValue - addressSectionAddressMValue).abs) > defaultTolerance
+    }
+
+    //Initial Part of The Section Splited
+    if ((trUAAsset.startAddressMValue > addressSection.startAddressMValue) && (verifyDefaultTolerance(trUAAsset.startAddressMValue, addressSection.startAddressMValue)) && (trUAAsset.startAddressMValue <= addressSection.endAddressMValue.head)) {
+      val newStartAddressSection = addressSection.copy(endAddressMValue = Some(trUAAsset.startAddressMValue))
+      calculateMeasures(roadAddress, newStartAddressSection) match {
+        case Some(measure) =>
+          val newTrAsset = TierekisteriTrafficSignData(newStartAddressSection.roadNumber, newStartAddressSection.roadPartNumber, newStartAddressSection.roadPartNumber, newStartAddressSection.track, newStartAddressSection.startAddressMValue, newStartAddressSection.startAddressMValue, roadSide, TRTrafficSignType.EndUrbanArea, defaultSpeedLimit.toString)
+          createLinearAsset(roadLink, roadAddress, newStartAddressSection, measure, newTrAsset)
+        case _ => None
+      }
+    }
+
+    //Middle Part of The Section Splited
+    if ((trUAAsset.startAddressMValue >= addressSection.startAddressMValue) && (trUAAsset.endAddressMValue <= addressSection.endAddressMValue.head)) {
+      val newMiddleAddressSection = addressSection.copy(startAddressMValue = trUAAsset.startAddressMValue, endAddressMValue = Some(trUAAsset.endAddressMValue))
+      calculateMeasures(roadAddress, newMiddleAddressSection) match {
```

Comment:
Maybe this part of the code can be put in some method in order to be use in the next if's